### PR TITLE
changed service type protocol

### PR DESCRIPTION
**Pull request by wigy-opensource-developer**
So @deuszex was working on this to allow services that were invented after the deployment of a LOC node to register into it. The string format can be specified outside the LOC protocol without changing the implementation.

### CODE REVIEW

**Review by wigy-opensource-developer — COMMENTED** [review]
As a courtesy for your fellow developers, please change your IDE settings to match the existing code, and force-push the changes, because most of the differences are just removal of orphaned whitespace on empty lines.

The protobuf generated code seems to be generated with a different version of the toolset. @izolyomi Could you please specify which version to use? That diff is also incomprehensibly big. Upgrading the toolset should be separated from the protocol change itself.

**Comment by izolyomi** [comment]
@wigy-opensource-developer I was using the protobuf-compiler 3.0 package shipped with recent Ubuntu releases. As far as I could tell, even the slightest change will cause huge diffs in the generated sources, so I'd consider this to be normal.

Unfortunately most of the time you have a different protobuf version (e.g. Ubuntu release 16.04LTS didn't have this package, so you had to install it manually) and code generated with different version usually doesn't compile, so you have to regenerate them. That's why there's a dedicated batch file for this near the sources.

**Comment by wigy-opensource-developer** [comment]
@deuszex Could you please try to regenerate the protobuf stuff with the 3.0 toolchain?

**Comment by izolyomi** [comment]
@deuszex Thanks for the pull request!

I don't mind removing trailing spaces from lines, seems I could have used better editor or settings myself. But the amount of changes makes it very hard to review, I had to manually force loading the diff of locnet.cpp which was hidden because of the size.

The real diffs (that I found :P) are nice and sound. There's only one futher change I'd suggest: there's no point of keeping the new `string<=>string From/ToProtoBuf()` converter in `messaging.hpp` which was previously `ServiceType<=>iop::locnet::service_type`.

File: src/spatialdb.cpp [hunk 1]
```diff
@@ -41,7 +41,7 @@ const vector<string> DatabaseInitCommands = {
     "  expiresAt    INT NOT NULL, " // Unix timestamp. NOTE consider implementing non expiring entries to have NULL here.
     "  location     POINT NOT NULL "
     ");"
-    
+
     "CREATE TABLE IF NOT EXISTS services ( "
     "  nodeId       TEXT NOT NULL, "
     "  serviceType  INT NOT NULL, "
```
Comment (wigy-opensource-developer) [inline src/spatialdb.cpp]: This has to be a TEXT so that we can save the string

**Review by wigy-opensource-developer — COMMENTED** [review]
Database schema change is also missing. And that also needs a migration script :frowning_face:

File: src/spatialdb.cpp [hunk 2]
```diff
@@ -44,7 +44,7 @@ const vector<string> DatabaseInitCommands = {
 
     "CREATE TABLE IF NOT EXISTS services ( "
     "  nodeId       TEXT NOT NULL, "
-    "  serviceType  INT NOT NULL, "
+    "  serviceType  TEXT NOT NULL, "
```
Comment (izolyomi) [inline src/spatialdb.cpp]: I see you have properly changed SQL variable binding types already, so the DB should work now.
But without this change I doubt that it was properly working before, did you perform any tests already?

**Comment by izolyomi** [comment]
As a new pull request (Feat/localservices #6) was created with a better version of the same thing, I close this old duplicate.

**Pull request closed without merging**